app/MMB/src/scenes/timeline/Timeline.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+
+import * as React from 'react';
+import { ScrollView } from 'react-native';
+import {
+  type RouteNamesType,
+  type NavigationType,
+} from '@kiwicom/mobile-navigation';
+import { DateUtils } from '@kiwicom/mobile-localization';
+
+import Events from './Events';
+
+type Props = {|
+  +navigation: NavigationType,
+|};
+
+export default class Timeline extends React.Component<Props> {
+  navigate = (key: RouteNamesType) => {
+    this.props.navigation.navigate({
+      routeName: key,
+      key: `key-${key}`,
+    });
+  };
+
+  render = () => {
+    const events = [
+      {
+        dateTime: new Date('2018-06-08T10:22:39.986Z'),
+        icon: { name: 'check', type: 'special' },
+        description: 'Boarding at gate 40',
+        comment: 'Closes 10 minutes before flight.',
+        location: 'in Brno',
+      },
+      {
+        dateTime: new Date('2018-06-07T10:22:39.986Z'),
+        icon: { name: 'check', type: 'ongoing' },
+        description: 'Boarding at gate 40',
+        comment: 'Closes 10 minutes before flight.',
+        location: 'in Brno',
+      },
+      {
+        dateTime: new Date('2018-06-08T14:22:39.986Z'),
+        icon: { name: 'flight-land', type: 'classic' },
+        description: 'Arrival',
+        comment: 'Enjoy the Eiffel Tower',
+        location: 'in Paris',
+      },
+    ];
+
+    const sortedEvents = events.sort((a, b) => a.dateTime - b.dateTime);
+
+    const enrichedEvents = sortedEvents.map((event, index) => ({
+      ...event,
+      first:
+        index === 0 ||
+        DateUtils.diffInDays(events[index - 1].dateTime, event.dateTime) !== 0,
```

Comment:
Why do we need to add this first bool here?

Reply:
And again - this is completely pointless. Create API first and your application will be clear. No sorting here, no magic boolean constants in your client...